### PR TITLE
Alola Team Skull and Aether Questline

### PR DESCRIPTION
**Pull request by Ultima1990**
Starring:
Aether Paradise gym.
Malie Garden dungeon.
Po Town dungeon.
Aether Foundation dungeon.
Altar of the Sunne and Moone gym.

And

Free master ball.

### CODE REVIEW

File: src/scripts/quests/QuestLineHelper.ts [hunk 1]
```diff
@@ -318,6 +318,39 @@ class QuestLineHelper {
         App.game.quests.questLines().push(plasmaUnovaQuestLine);
     }
 
+    public static createSkullAetherAlolaQuestLine() {
+        const skullAetherAlolaQuestLine = new QuestLine('Eater of Light', 'A dangerous Pokémon from another world threatens the Alola region.');
+
+        const clearAetherParadiseGym = new CustomQuest(1, 0, 'A strange creature has appeared in Aether Paradise. Make it go away. Clear the Aether Paradise gym.', () => App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Aether Paradise')](), App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Aether Paradise')]());
```
Comment (RedSparr0w) [inline src/scripts/quests/QuestLineHelper.ts:324]: Do we need the extra `gymsDefeated` at the end for this?
It should automatically be based on your current progress (based on the value that the focus returns)
```suggestion
        const clearAetherParadiseGym = new CustomQuest(1, 0, 'A strange creature has appeared in Aether Paradise. Make it go away. Clear the Aether Paradise gym.', () => App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Aether Paradise')]());
```

File: src/scripts/quests/QuestLineHelper.ts [hunk 2]
```diff
@@ -318,6 +318,39 @@ class QuestLineHelper {
...
+            App.game.pokeballs.gainPokeballs(GameConstants.Pokeball.Masterball, 1, false);
+            Notifier.notify({
+                title: skullAetherAlolaQuestLine.name,
+                message: 'You found a Masterball!',
+                type: NotificationConstants.NotificationOption.success,
+                timeout: 3e4,
+            });
+        };
+
+        const clearAltaroftheSunnandMooneGym = new CustomQuest(1, AltaroftheSunnandMooneGymReward, 'Stop the Eather of Light from absorbing all light in Alola. Clear the Altar of the Sunne and Moone gym.', () => App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Altar of the Sunne and Moone')](), App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Altar of the Sunne and Moone')]());
```
Comment (RedSparr0w) [inline src/scripts/quests/QuestLineHelper.ts:347]: Same thing as above comment
```suggestion
        const clearAltaroftheSunnandMooneGym = new CustomQuest(1, AltaroftheSunnandMooneGymReward, 'Stop the Eather of Light from absorbing all light in Alola. Clear the Altar of the Sunne and Moone gym.', () => App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Altar of the Sunne and Moone')]());
```

File: src/scripts/quests/QuestLineHelper.ts [hunk 3]
```diff
@@ -318,6 +318,39 @@ class QuestLineHelper {
...
+        console.log('clears:', App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Altar of the Sunne and Moone')]());
```
Comment (RedSparr0w) [inline src/scripts/quests/QuestLineHelper.ts:348]: A wild `console.log` appeared
```suggestion
```

File: src/scripts/quests/QuestLineHelper.ts [hunk 4]
```diff
@@ -318,6 +318,39 @@ class QuestLineHelper {
         App.game.quests.questLines().push(plasmaUnovaQuestLine);
     }
 
+    public static createSkullAetherAlolaQuestLine() {
+        const skullAetherAlolaQuestLine = new QuestLine('Eater of Light', 'A dangerous Pokémon from another world threatens the Alola region.');
+
+        const clearAetherParadiseGym = new CustomQuest(1, 0, 'A strange creature has appeared in Aether Paradise. Make it go away. Clear the Aether Paradise gym.', () => App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Aether Paradise')](), App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Aether Paradise')]());
+        console.log('clears:', App.game.statistics.gymsDefeated[GameConstants.getGymIndex('Aether Paradise')]());
```
Comment (RedSparr0w) [inline src/scripts/quests/QuestLineHelper.ts:325]: A wild `console.log` appeared
```suggestion
```

**Review by RedSparr0w — APPROVED** [review]
Looks good, just a couple minor comments.
Thanks for the PR! 👍

File: src/scripts/quests/QuestLineHelper.ts (hunk 1 above)
Comment (Ultima1990) [reply src/scripts/quests/QuestLineHelper.ts:324]: NGL I just copied it down from the Plasma one.

**Pull request merged**